Automattic/wp-calypso
# Update: refactor product store to share context

#### Proposed Changes

This PR moves `storeItemInfo` to the context wrapped over the main components for easier reuse.

#### Testing Instructions

* Run `git fetch && git checkout update/jetpack-product-lightbox-product-data`
  * Run `yarn start-jetpack-cloud`
  * Go to http://jetpack.cloud.localhost:3000/pricing?flags=jetpack/pricing-page-product-lightbox

**OR**

* use the jetpack cloud live link and goto `/pricing?flags=jetpack/pricing-page-product-lightbox

* Verify that page still operates as expected. No changes should be seen here. 
* Open Product Lightbox by clicking `More about ...` button on a product card. Checkout button should work correctly with proper URL


*

#### Pre-merge Checklist






Related to #

1202796695664022-as-1202796695664087/f
